### PR TITLE
shipyard: init at 0.1.17

### PR DESCRIPTION
**Pull request by cpcloud**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "shipyard-run";
+    repo = pname;
+    sha256 = "035a12sl1my1fgvjab9la6j0r79ph9kkrycqrzgg08iz6kn23pjz";
+  };
+
+  subPackages = [ "." ];
+  vendorSha256 = "03qhvrpm6d1h6501885zac2gxa9q449rs7y82r2arhz6mky4p2lw";
+  buildFlagsArray = ''
```
Comment (doronbehar) [inline pkgs/tools/virtualization/shipyard/default.nix:15]: Try:

```suggestion
  buildFlagsArray = [
```

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pname = "shipyard";
+  version = "0.0.37";
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "shipyard-run";
+    repo = pname;
+    sha256 = "035a12sl1my1fgvjab9la6j0r79ph9kkrycqrzgg08iz6kn23pjz";
+  };
+
+  subPackages = [ "." ];
```
Comment (doronbehar) [inline pkgs/tools/virtualization/shipyard/default.nix:13]: What happens if you leave this out?

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  vendorSha256 = "03qhvrpm6d1h6501885zac2gxa9q449rs7y82r2arhz6mky4p2lw";
```
Comment (doronbehar) [inline pkgs/tools/virtualization/shipyard/default.nix:14]: Please move this line to be right below `src`.

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "shipyard";
+  version = "0.0.37";
+  src = fetchFromGitHub {
```
Comment (doronbehar) [inline pkgs/tools/virtualization/shipyard/default.nix:6]:
```suggestion

  src = fetchFromGitHub {
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/tools/virtualization/shipyard/default.nix (hunk 2 above)
Comment (cpcloud) [reply pkgs/tools/virtualization/shipyard/default.nix]: Done.

File: pkgs/tools/virtualization/shipyard/default.nix (hunk 3 above)
Comment (cpcloud) [reply pkgs/tools/virtualization/shipyard/default.nix:13]: Done.

**Comment by cpcloud** [comment]
Looks like a new release happened since I opened this PR, going to give this one a new title and rebase+squash.

**Review by doronbehar — APPROVED** [review]
Passes `nixpkgs-review` build and `--help` test.

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    "-ldflags="
+    "-X"
+    "main.version=${version}"
+  ];
+  meta = {
+    description = "Shipyard is a tool for building modern cloud native development environments";
+    homepage = "https://shipyard.run";
+    license = lib.licenses.mpl20;
+    maintainers = with lib.maintainers; [ cpcloud ];
+    platforms = lib.platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/shipyard/default.nix:25]:
```suggestion
```
That is not required.

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  buildFlagsArray = [
+    "-ldflags="
+    "-X"
+    "main.version=${version}"
+  ];
+  meta = {
+    description = "Shipyard is a tool for building modern cloud native development environments";
+    homepage = "https://shipyard.run";
+    license = lib.licenses.mpl20;
+    maintainers = with lib.maintainers; [ cpcloud ];
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/shipyard/default.nix:24]:
```suggestion
  meta = with lib; {
    description = "Shipyard is a tool for building modern cloud native development environments";
    homepage = "https://shipyard.run";
    license = licenses.mpl20;
    maintainers = with maintainers; [ cpcloud ];
```

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "shipyard-run";
+    repo = pname;
+    sha256 = "0ycawkrbqrcivm7vxvqwyl6mlz7plbl5q2nryvhkr1l3wk4zbnk9";
+  };
+  vendorSha256 = "0bpx53b34a1g2gypiwmxz1d6m52dbi6xzh3diyr39v4z96f1c5zp";
+
+  buildFlagsArray = [
+    "-ldflags="
+    "-X"
+    "main.version=${version}"
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/shipyard/default.nix:18]:
```suggestion
    "-ldflags="
    "-X"
    "-s"
    "-w"
    "main.version=${version}"
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Please disable the failing test:
```
panic: mkdir /homeless-shelter: permission denied [recovered]
    panic: mkdir /homeless-shelter: permission denied

goroutine 15 [running]:
testing.tRunner.func1.1(0x1b2f0c0, 0xc000133890)
    testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc000a6a780)
    testing/testing.go:1075 +0x41a
panic(0x1b2f0c0, 0xc000133890)
    runtime/panic.go:969 +0x1b9
github.com/shipyard-run/shipyard/pkg/utils.ShipyardTemp(0x52af11, 0xc00096f380)
    github.com/shipyard-run/shipyard/pkg/utils/utils.go:129 +0xe6
github.com/shipyard-run/shipyard/pkg/clients.TestContainerCreatesDirectoryForVolume(0xc000a6a780)
    github.com/shipyard-run/shipyard/pkg/clients/docker_tasks_container_create_test.go:293 +0x4c
testing.tRunner(0xc000a6a780, 0x1e14418)
    testing/testing.go:1123 +0xef
created by testing.(*T).Run
    testing/testing.go:1168 +0x2b3
FAIL    github.com/shipyard-run/shipyard/pkg/clients    0.047s
FAIL
```

**Comment by SuperSandro2000** [comment]
@cpcloud Please fix the merge conflict

**Comment by cpcloud** [comment]
@SuperSandro2000 Fixed!

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X"
+    "main.version=${version}"
+  ];
+
+  doCheck = false;
```
Comment (doronbehar) [inline pkgs/tools/virtualization/shipyard/default.nix:23]: This is the default, so either comment why you want to make sure tests will not run, or remove it.

File: pkgs/tools/virtualization/shipyard/default.nix [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "13cp7qpxchnyxdm26xwdcp557nj16f4h8vlj0p4h79z5g7pcklln";
+  };
+  vendorSha256 = "0gib9s09lz91wawbms9zq4wc5k6bdxfzpxm8q92h0bsjw1bj1hzs";
+
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X"
+    "main.version=${version}"
```
Comment (doronbehar) [inline pkgs/tools/virtualization/shipyard/default.nix:20]: Nit: usually it is done differently. See https://nixos.org/manual/nixpkgs/unstable/#ex-goBuildFlags-noarray

File: pkgs/tools/virtualization/shipyard/default.nix (hunk 8 above)
Comment (cpcloud) [reply pkgs/tools/virtualization/shipyard/default.nix:21]: Added a comment.

**Comment by cpcloud** [comment]
@doronbehar On it.

**Comment by doronbehar** [comment]
Result of `nixpkgs-review pr 94561` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>shipyard</li>
  </ul>
</details>

**Pull request merged**